File: examples/sync/bitmex_public.py
import dcex


def main() -> None:
    client = dcex.bitmex()

    orderbook = client.get_orderbook(product_symbol="XBT-USDT-SWAP")
    print(orderbook)

    ticker = client.get_ticker(symbol="XBT-USDT-SWAP", binSize="1m", count=5)
    print(ticker)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/bitmex_public.rs
use std::time::Duration;

use dcex::exchanges::bitmex::BitmexClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = BitmexClient::public(Duration::from_secs(10))?;
    let response = client.get_orderbook("XBT-USDT-SWAP").await?;
    println!("{}", response.data);
    Ok(())
}
